File: 1252. Cells with Odd Values in a Matrix.js
// 1252. Cells with Odd Values in a Matrix
m = 2; n = 3; indices = [[0,1],[1,1]]

function oddCells(m, n, indices) {
  let matrix = Array.from({ length: m }, () => Array(n).fill(0));
  
  indices.forEach(([r, c]) => {
      for (let i = 0; i < n; i++) {
          matrix[r][i] += 1;
      }
      for (let i = 0; i < m; i++) {
          matrix[i][c] += 1;
      }
  });

  return matrix.flat().filter(c => c % 2 !== 0).length;
}
console.log(oddCells(m, n, indices));  // Output: 6

File: 1252. Cells with Odd Values in a Matrix.rb
#* 1252. Cells with Odd Values in a Matrix
m = 2; n = 3; indices = [[0,1],[1,1]]
# @param {Integer} m
# @param {Integer} n
# @param {Integer[][]} indices
# @return {Integer}
def odd_cells(m, n, indices) #m=col|n=row
  matrix = Array.new(m){Array.new(n){0}}
  indices.each {|rc|
    (0...n).each do |i| 
      matrix[rc[0]][i] += 1
    end
    (0...m).each do |i|
      matrix[i][rc[1]] += 1
    end
  }
  matrix.flatten.count{|c| c.odd?}
end
p odd_cells(m, n, indices)